src/Test.Utilities/CodeFixTestBase.cs

```diff
@@ -26,43 +24,72 @@ public abstract class CodeFixTestBase : DiagnosticAnalyzerTestBase
 
         protected void VerifyCSharpUnsafeCodeFix(string oldSource, string newSource, int? codeFixIndex = null, bool allowNewCompilerDiagnostics = false, bool onlyFixFirstFixableDiagnostic = false)
         {
-            VerifyFix(LanguageNames.CSharp, GetCSharpDiagnosticAnalyzer(), GetCSharpCodeFixProvider(), oldSource, newSource, codeFixIndex, allowNewCompilerDiagnostics, onlyFixFirstFixableDiagnostic, DefaultTestValidationMode, true);
+            var runner = new CodeFixRunner(GetCSharpDiagnosticAnalyzer(), GetCSharpCodeFixProvider(), DefaultTestValidationMode);
+            VerifyFix(LanguageNames.CSharp, runner, new[] { oldSource }, new[] { newSource }, codeFixIndex, allowNewCompilerDiagnostics, onlyFixFirstFixableDiagnostic, allowUnsafeCode: true);
         }
 
         protected void VerifyCSharpFix(string oldSource, string newSource, int? codeFixIndex = null, bool allowNewCompilerDiagnostics = false, bool onlyFixFirstFixableDiagnostic = false, TestValidationMode validationMode = DefaultTestValidationMode)
         {
-            VerifyFix(LanguageNames.CSharp, GetCSharpDiagnosticAnalyzer(), GetCSharpCodeFixProvider(), oldSource, newSource, codeFixIndex, allowNewCompilerDiagnostics, onlyFixFirstFixableDiagnostic, validationMode, false);
+            var runner = new CodeFixRunner(GetCSharpDiagnosticAnalyzer(), GetCSharpCodeFixProvider(), validationMode);
```

Comment:
Any reason we can't move the instantiation of the `CodeFixRunner` into `VerifyFix`? Seems unnecessary to duplicate it in all helpers.